src/aws-cpp-sdk-core/source/utils/logging/FormattedLogSystem.cpp

```diff
@@ -17,44 +17,56 @@
 using namespace Aws::Utils;
 using namespace Aws::Utils::Logging;
 
-static Aws::String CreateLogPrefixLine(LogLevel logLevel, const char* tag)
+static const char* GetLogPrefix(const LogLevel logLevel)
 {
-    Aws::StringStream ss;
-
-    switch(logLevel)
+    switch (logLevel)
     {
-        case LogLevel::Error:
-            ss << "[ERROR] ";
+        case LogLevel::Off:
+            assert(!"This should never happen.");
             break;
-
+        case LogLevel::Error:
+            return "[ERROR] ";
         case LogLevel::Fatal:
-            ss << "[FATAL] ";
-            break;
-
+            return "[FATAL] ";
         case LogLevel::Warn:
-            ss << "[WARN] ";
-            break;
-
+            return "[WARN] ";
         case LogLevel::Info:
-            ss << "[INFO] ";
-            break;
-
+            return "[INFO] ";
         case LogLevel::Debug:
-            ss << "[DEBUG] ";
-            break;
-
+            return "[DEBUG] ";
         case LogLevel::Trace:
-            ss << "[TRACE] ";
-            break;
-
-        default:
-            ss << "[UNKOWN] ";
-            break;
+            return "[TRACE] ";
     }
+    return "[UNKNOWN] ";
+}
 
-    ss << DateTime::Now().CalculateGmtTimeWithMsPrecision() << " " << tag << " [" << std::this_thread::get_id() << "] ";
-
-    return ss.str();
+static Aws::String CreateLogPrefixLine(LogLevel logLevel, const char* tag, const size_t statementSize = 80)
+{
+    Aws::String prefix;
+    static const size_t THREAD_ID_LEN = [](){
+        Aws::StringStream strStr;
+        strStr << std::this_thread::get_id();
+        return strStr.str().size();
```

Comment:
this runs only once per application instance, right?

Is it guaranteed that every thread's ID is the same length? If this were just an integer on some platform, I could imagine different threads having different length integers. I guess it's still safe, since this is just for a `reserve()` call...

Simpler to just have a hard-coded high guess? Or overestimate somewhere and just like, ` + 10` to prefixLen?